src/ipv6.cpp

```diff
@@ -169,6 +212,33 @@ uint32_t IPv6::get_padding_size(const ext_header& header) {
     return padding == 0 ? 0 : (8 - padding);
 }
 
+std::vector<IPv6::header_option_type> IPv6::parse_header_options(const uint8_t* data, size_t size)
+{
+    Memory::InputMemoryStream stream(data, size);
+    std::vector<header_option_type> options;
+
+    while (stream.size() > 0) {
+        try {
+            uint8_t option = stream.read<uint8_t>();
+            if (option == OptionType::PAD_1) {
+                continue;
+            }
+            uint8_t size = stream.read<uint8_t>();
+            if (size > stream.size()) {
+                throw invalid_ipv6_extension_header();
+            }
+            if (option != PAD_N) {
+                std::vector<uint8_t> value(stream.pointer(), stream.pointer() + size);
+                options.push_back(std::make_pair(option, value));
```

Comment:
So this sucks because I'm trying to keep the codebase both C++03 and C++11 compliant. If this was C++11 you would do:

```c++
options.push_back(make_pair(option, move(value)));
```

So you avoid creating an unnecessary copy of the vector. So in this case you can either have some ugly `ifdef TINS_IS_CXX11` and use move, otherwise do what this is moving or you can just move this into a single expression:

```c++
options.push_back(make_pair(option, vector<uint8_t>(stream.pointer(), stream.pointer() + size)));
```

Possibly having to split that in multiple lines if it's too long.

Reply:
I thought a bit about that actually and was pretty sure the compiler could probably optimize the copy away, but I never tested that.

Anyway since this library has to support C++03, I much prefer your second option over ugly #ifdefs, so I'll go with that.

We spotted.